zephyr/esp32c2/src/wifi/esp_wifi_adapter.c

```diff
@@ -0,0 +1,1062 @@
+/*
+ * Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define CONFIG_POSIX_FS
+
+#include "esp_wifi.h"
+#include "stdlib.h"
+#include "string.h"
+#include "esp_private/wifi.h"
+#include "soc/soc.h"
+#include "soc/dport_access.h"
+#include "soc/wdev_reg.h"
+#include "soc/soc_caps.h"
+#include "soc/rtc.h"
+#include "soc/system_reg.h"
+#include "esp_log.h"
+#include "esp_attr.h"
+#include "esp_private/wifi_os_adapter.h"
+#include "esp_system.h"
+#include "esp_timer.h"
+#include "os.h"
+#include "esp_wpa.h"
+#include "esp_private/periph_ctrl.h"
+#include "esp_phy_init.h"
+#include <soc/syscon_reg.h>
+#include <rom/efuse.h>
+#include <rom/rtc.h>
+#include <rom/ets_sys.h>
+#include <riscv/interrupt.h>
+#include "esp32c3/rom/ets_sys.h"
+#include "esp_mac.h"
+#include "wifi/wifi_event.h"
+
+#include <soc.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/random/random.h>
+#include <zephyr/drivers/interrupt_controller/intc_esp32c3.h>
+#include "zephyr_compat.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(esp32_wifi_adapter, CONFIG_WIFI_LOG_LEVEL);
+
+ESP_EVENT_DEFINE_BASE(WIFI_EVENT);
+
+static void esp_wifi_free(void *mem);
+
+static void *wifi_msgq_buffer;
+
+static struct k_thread wifi_task_handle;
+
+uint64_t g_wifi_feature_caps =
+#if CONFIG_ESP_WIFI_ENABLE_WPA3_SAE
```

Comment:
Is this `g_wifi_features` list checked against C2 features?

Reply:
Actually not. Can we remove this structure as it is present in `wifi_init.c`?

Reply:
Sorry, bad question. FTM support is okay.
WPA3 and SPIRAM seem standard to me, but I couldn't be 100% sure.